### ISSUE TITLE
app-registry plugin for Helm 3

### ISSUE THREAD

**Issue by pho-enix** [app-registry/appr#102]
These days Helm v3 was released. The plugin does not work there. After installing to the plugin directory, `helm plugin list` does not list it and the command can also not be called.

Furthermore with the OCI support of Helm v3 the `helm registry` subcommand will already be in use by Helm itself (see: https://v3.helm.sh/docs/topics/registries/). Thus a rename of the plugin might be required.

**Comment by pho-enix** [comment]
I forked the plugin:
https://github.com/pho-enix/helm-quay

**Comment by ant31** [comment]
would you be able to do a PR ?

**Comment by pho-enix** [comment]
Is this project still active?

**Comment by ant31** [comment]
Not very active but still used

**Comment by ogruene** [comment]
Hi @ant31, @pho-enix,
we are currently also evaluating possibilities on how we can use/modify the existing helm Quay plugin with Helm v3. We might also create a PR in [plugin repo](https://github.com/app-registry/appr-helm-plugin) if it's necessary. Is there a realistic chance that the PR will be approved and merged? We are aware that this a breaking change as the `helm registry` will then no longer be available for use with Helm v2 and must be replaced with `helm quay`. We would highly appreciate any feedback on this.
Thanks.

Oliver Grüneberg <<oliver.grueneberg@daimler.com>>, Daimler TSS GmbH
[Legal Imprint](https://github.com/Daimler/daimler-foss/blob/master/LEGAL_IMPRINT.md)

**Comment by ant31** [comment]
Hi, 

I have only few overview of HelmV3, also I know they integrate directly an OCI registry and eventually would become compatible nativily at some point (cc @jzelinskie) 

I'll create a separate repo: appr-helmV3-plugin, so there is no issue with compatibility. I'll merge any PR submitted to it. 

Thank you.

**Comment by ogruene** [comment]
Sounds good for us although the new Helm plugin should still work with Helm v2 with `helm quay` instead of `helm registry`, so the repo name might be misleading ;-)

Thank you.

**Comment by ant31** [comment]
https://github.com/app-registry/quay-helmv3-plugin

**Comment by pho-enix** [comment]
@ant31 You can simply fork (which is a fork of your repo):
https://github.com/pho-enix/helm-quay

The change to `helm quay` is included plus some fixes and improvements for plugin installation.

What do you think?

@ogruene any additional functionality that you need?

**Comment by ogruene** [comment]
Hi @pho-enix, @ant31,
I don't think that we need additional functionality. I'm preparing right now a PR for the new v3 repository. It is also OK for us if @ant31 takes your repository as base for the new repo. If we need changes we can still do a PR. Our problem is that we need a **quick** solution ;-)
So I will go further on preparing the PR ...

@pho-enix: As far as I see your repo is not a fork but furthermore a copy/clone, is it?

**Comment by pho-enix** [comment]
> @pho-enix: As far as I see your repo is not a fork but furthermore a copy/clone, is it?

Yeah. Sorry for that. So you'd have to manually apply my changes again. But it's acually not a lot of diff.

**Comment by ogruene** [comment]
Hi @ant31,
the PR was opened last week. I would highly appreciate if you can review and merge the PR ;-)

Thank you very much.
